src/fw-update/fw-update-device.cpp

```diff
@@ -112,6 +139,19 @@ namespace librealsense
         return false;
     }
 
+    float update_device::compute_progress(float progress, float start, float end, float threshold) const
+    {
+        return start + (ceil(progress * threshold) / threshold) * (end - start) / 100.f;
+    }
+
+    bool update_device::wait_for_manifest_completion(std::shared_ptr<platform::usb_messenger> messenger, const rs2_dfu_state state, 
+        std::chrono::seconds timeout_seconds, rs2_update_progress_callback_sptr update_progress_callback) const
+    {
+        // used for devices which get the progress percentage in the GET_DFU_STATUS call 
+
+        return true;
```

Comment:
Why do we always return true?

Reply:
The method wait_for_manifest_completion in the fw-update-device class just returns true
In the class d500-fw-update-device, it does all the process of polling the progress percentage of the flash updating.
Just using the inheritance to do the needed actions acc to the device.